orgbee/src/app/login/login.component.ts

```diff
@@ -1,10 +1,51 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrl: './login.component.css'
+  styleUrls: ['./login.component.css']
 })
-export class LoginComponent {
+export class LoginComponent implements OnInit {
+  loginForm: FormGroup;
+  signupForm: FormGroup;
+  showSignup: boolean = false;
 
+  ngOnInit() {
+    this.loginForm = new FormGroup({
+      email: new FormControl('', [Validators.required, Validators.email]),
+      password: new FormControl('', Validators.required),
+    });
+
+    this.signupForm = new FormGroup({
+      firstName: new FormControl('', Validators.required),
+      lastName: new FormControl('', Validators.required),
+      studentNumber: new FormControl('', Validators.required),
+      email: new FormControl('', [Validators.required, Validators.email]),
+      bday: new FormControl('', Validators.required),
+      gender: new FormControl('', Validators.required),
+      password: new FormControl('', [Validators.required, Validators.minLength(6)]),
+      confirmPassword: new FormControl('', Validators.required)
+    });
+  }
+
+  onSubmit() {
+    if (!this.loginForm.valid) return;
+
+    console.log(this.loginForm.value);
+  }
+
+  showSignupPopup() {
+    this.showSignup = true;
+  }
```

Comment:
The popup won't show up on my side probably because of the href. The solution I found is to add **return false;** after **this.showSignup = true;** to prevent the default action.

`  showSignupPopup() {`
`    this.showSignup = true;`
`    return false;`
`  }`